[DOCS] Adds UI related limitation to configuring aggs docs

szabosteve opened the pull request:
## Overview

This PR adds a UI related limitation to the `Requirements and limitations` section of the Aggregating data for faster performance doc. The limitation is about aggregation types in datafeeds that are not supported by the Single Metric Viewer in Kibana.

**Applies to 7.11 and above.** 
(https://github.com/elastic/elasticsearch/pull/65190 handles 7.10 and below.)

### Preview

[Requirements and limitations](https://elasticsearch_65184.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-configuring-aggregation.html#aggs-limits-dfeeds)

szabosteve committed:
[DOCS] Adds UI related limitation to configuring aggs docs.

elasticmachine commented:
Pinging @elastic/ml-core (:ml)

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

peteharverson commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,10 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
 your detectors use <<ml-count-functions,count>> or <<ml-rare-functions,rare>>
 functions, set the interval to the same value as the bucket span.
 
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
```
This limitation also applies to the anomaly charts in the Anomaly Explorer.

szabosteve committed:
[DOCS] Addresses feedback.

szabosteve commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,11 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
 
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
+plot is not enabled for the {anomaly-job}, then neither the **Single Metric 
+Viewer** nor the ***Anomaly Explorer** in {kib} can plot and display an anomaly 
```
```suggestion
Viewer** nor the **Anomaly Explorer** in {kib} can plot and display an anomaly 
```

szabosteve committed:
Update docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc

qn895 commented:
Currently we also don't support configurations where the aggregation interval is not the same as the bucket span in the Single Metric Viewer. For example: 

```javascript
// datafeed config
  "aggregations": {
    "buckets": {
      "date_histogram": {
        "field": "@timestamp",
        "time_zone": "UTC",
        "fixed_interval": "15m"
      },
      "aggregations": {
        "@timestamp": {
          "max": {
            "field": "@timestamp"
          }
        },
		...
      }
    }
  },
```

```
// job config
{
  "job_type": "anomaly_detector",
  "description": "",
  "analysis_config": {
    "bucket_span": "30m", // bucket span different from buckets.date_historgram.fixed_interval
    "summary_count_field_name": "doc_count",
	...
	}
  ...
}
```

szabosteve committed:
[DOCS] Addresses feedback.

szabosteve committed:
[DOCS] Minor changes.

szabosteve committed:
[DOCS] Fixes markup.

szabosteve committed:
[DOCS] Fixes typo.

szabosteve commented:
@qn895 Thanks so much for the info! I added the above-mentioned case to the docs as well. I haven't added the code snippet though because I didn't want to add an example of how to do it wrong. On the other hand, I see the value of such an example, too. I added an extra explanatory sentence about the config to make the case clearer without adding the snippet. What do you think?

peteharverson commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,15 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
+plot is not enabled for the {anomaly-job}, then neither the **Single Metric 
+Viewer** nor the **Anomaly Explorer** in {kib} can plot and display an anomaly 
+chart for the job. Single Metric Viewer doesn't support configurations where 
```
Sorry, this needs more editing as this issue affects the charts in both the Anomaly Explorer and Single Metric Viewer. I'm not sure how best to phrase this - charts are displayed, but the values plotted in the charts don't match up unless the job bucket span is the same as the interval used for the date_histogram aggregation. For example, here the bucket_span is `30m` and the date_histogram aggregation interval is `15m`. The result is the data in the chart is twice the actual value:

![image](https://user-images.githubusercontent.com/7405507/99655168-44cefc80-2a53-11eb-96ef-89d23f2ed989.png)

szabosteve committed:
[DOCS] Makes limitation more accurate.

peteharverson reviewed (approved):
Latest edit LGTM.

lcawl commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
 your detectors use <<ml-count-functions,count>> or <<ml-rare-functions,rare>>
 functions, set the interval to the same value as the bucket span.
 
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
```
In case people don't notice the formatting, it might be nice to qualify this:
```suggestion
If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms` aggs>> and model 
```

lcawl commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
 functions, set the interval to the same value as the bucket span.
 
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
+plot is not enabled for the {anomaly-job}, then neither can the **Single Metric 
```
IMO we can usually leave out "then" in cases like this:
```suggestion
plot is not enabled for the {anomaly-job}, neither the **Single Metric 
```

lcawl commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
 
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
+plot is not enabled for the {anomaly-job}, then neither can the **Single Metric 
+Viewer** nor the **Anomaly Explorer** in {kib} plot and display an anomaly 
```
IMO we don't need to point out they're in Kibana here, but am fine with being overruled
```suggestion
Viewer** nor the **Anomaly Explorer** can plot and display an anomaly 
```

lcawl commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+chart for the job.
+
+When the aggregation interval of the {dfeed} and the bucket span of the job 
+don't match, the values of the chart plotted in both the Single Metric Viewer 
```
Adding missing bold:
```suggestion
don't match, the values of the chart plotted in both the **Single Metric Viewer** 
```

lcawl commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms`>> and model 
+plot is not enabled for the {anomaly-job}, then neither can the **Single Metric 
+Viewer** nor the **Anomaly Explorer** in {kib} plot and display an anomaly 
+chart for the job.
```
> chart for the job.

As a reader, this makes me wonder whether an error is returned or the charts are just blank. If you think it's not overkill, it might be nice to tell them what they'll experience.

lcawl commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+
+When the aggregation interval of the {dfeed} and the bucket span of the job 
+don't match, the values of the chart plotted in both the Single Metric Viewer 
+and the Anomaly Explorer differs from the actual values of the job. To avoid 
```
Since I think it's the the "values" (plural) that differ, we should drop the "s" here:
```suggestion
and the **Anomaly Explorer** differ from the actual values of the job. To avoid 
```

szabosteve committed:
Apply suggestions from code review

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

szabosteve committed:
[DOCS] Addresses feedback.

szabosteve commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms` aggs>> and 
+model plot is not enabled for the {anomaly-job}, neither the **Single Metric 
+Viewer** nor the **Anomaly Explorer** can plot and display an anomaly 
+chart for the job. In this cases, the charts are blank.
```
```suggestion
chart for the job. In these cases, the charts are blank.
```

szabosteve committed:
Update docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc

qn895 commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,18 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms` aggs>> and 
+model plot is not enabled for the {anomaly-job}, neither the **Single Metric 
+Viewer** nor the **Anomaly Explorer** can plot and display an anomaly 
+chart for the job. In these cases, the charts are blank.
```
My apologies for another change, for 7.11, in these cases, the charts will not be visible and there will be call outs explaining why.

szabosteve replied:
Modified the text accordingly via 5148e84.

szabosteve committed:
[DOCS] MInor edits.

szabosteve committed:
[DOCS] Further edits.

szabosteve commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,23 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+If your <<aggs-dfeeds,{dfeed} uses aggregations with nested `terms` aggs>> and 
+model plot is not enabled for the {anomaly-job}, neither the **Single Metric 
+Viewer** nor the **Anomaly Explorer** can plot and display an anomaly 
+<<<<<<< HEAD
```
```suggestion
```

szabosteve commented on docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc:
```diff
@@ -45,6 +45,23 @@ finer, more granular time buckets, which are ideal for this type of analysis. If
+<<<<<<< HEAD
+chart for the job. In this cases, the charts are not visible and an explanatory 
+message is shown.
+=======
+chart for the job. In these cases, the charts are blank.
+>>>>>>> c7ee66b56a48246a14543ad24dc6b691a1c9a226
```
```suggestion
```

szabosteve committed:
Apply suggestions from code review
